### ISSUE TITLE
synchronize bookmarks

### ISSUE THREAD

**Issue by trazomtg** [QupZilla/qupzilla#2094]
Hi,

as you see, i am a new user of Qupzilla.
I'd like toknow how to synchronize bookmarks on different laptops.

Thanks
T

**Comment by klundry** [comment]
The bookmarks.json file is in your home directory and I have tested just copying it to a different machine and it works fine.  It should be possible to setup a sync solution that syncs this file out to other machines when it changes.  It's not ideal though and would be nice to have a way to do that built-in to the browser.

**Comment by Martii** [comment]
>  would be nice to have a way to do that built-in to the browser.

Transferring can be achieved via Bookmarks &rarr; Organize Bookmarks which has an import and export function as spotlighted below:

![bookmarks](https://cloud.githubusercontent.com/assets/114709/19613855/a16c6ac4-97ac-11e6-9224-5eaba87d1182.png)

**Comment by klundry** [comment]
Yes, transferring bookmarks is there.  I think the functionality we are looking for is the ability to keep the bookmarks in sync across multiple devices.  I would rather not have to manually export my bookmarks, find a way to transfer the file to other devices and then import them again.  That is not convenient to have to do it that way every time I add or remove a bookmark.

**Comment by cranes-bill** [comment]
You can manually transfer **bookmarks.json**.This file is located in _profile directory_.In addition, you can also transfer **browsdata.db**, which contains _history_, _passwords_, etc. You can even transfer all of the _profile directory_. But i can propose to make a symbolic link, for example:
Move  **bookmarks.json** to _sd card_, then create a symbolic link to this file.

**Comment by Strit** [comment]
I just started using Qupzilla and was looking for this functionality.
Would be great for an in-browser syncronization of the bookmarks acroos machines.

**Comment by Gibbz** [comment]
yes id love to see sync support. Ive just lost my qupzilla install and all my bookmarks went down the drain too :(

**Comment by micbsv** [comment]
This may be implemented very easily:
- Add the path to the profile folder to the QupZilla settings to make it configurable
- Put the profile folder into DropBox folder
That's it!

**Comment by Strit** [comment]
> Put the profile folder into DropBox folder
> That's it!

Relying on other services for bookmark syncing is not a solution, but a workaround.
The solution would be that Qupzilla itself could handle it.

**Comment by micbsv** [comment]
Agree, but personally, I would really appreciate even for this little work around. QupZilla is an awesome lightweight browser and only the lack of the syncing holds me on Chrome.

**Comment by Gibbz** [comment]
Actually. It would be handy to have an option to sync to a separate
directory and also to a sftp/ftp location.

**Comment by luchmhor** [comment]
Would it be possible to use FF Sync? Epiphany (Gnome Web) is utilizing FF Sync, although it uses webkit as engine:

https://blogs.gnome.org/mcatanzaro/2017/08/09/on-firefox-sync/

Benefits:
- sync of various things (e.g. bookmarks, passwords)
- existing infrastructure
- no need for self-baked sync scripts, file copying (never really working consistently)
- maybe easy implementation due to the work already done for gnome-web
